### ISSUE TITLE
Create an scaleway_instance_server data source

### ISSUE THREAD

**Issue by lnovara** [terraform-providers/terraform-provider-scaleway#102]
Add a server data source to query for servers by name or by tag.

I was thinking of something along the lines of:
```hcl
data "scaleway_server" "server_by_name" {
  name = "my_server"
}

data "scaleway_server" "server_by_tags" {
  tags_filter = ["foo", "bar"]
}
```

What do you think about this feature? If there is any interest I can provide a PR.

**Comment by nicolai86** [comment]
I think this can be a useful feature especially in combination with the cloudinit support added in 1.8.

Let me know if you need help with anything while working on the PR :)

**Comment by kindermoumoute** [comment]
fixed by #350
